src/modm/platform/clock/cortex/modm_precision_clock.cpp

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright (c) 2017-2018, Niklas Hauser
+ *
+ * This file is part of the modm project.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+// ----------------------------------------------------------------------------
+
+#include <modm/architecture/interface/clock.hpp>
+#include <modm/architecture/interface/precision_clock.hpp>
+#include <modm/platform/device.hpp>
+
+template< typename UsTimeType >
+UsTimeType
+modm::PrecisionClock::now()
+{
+	UsTimeType usValue(((SysTick->LOAD - SysTick->VAL) * 1000) / SysTick->LOAD);
+	return modm::Clock::now().getTime() + usValue;
```

Comment:
Should you not multiply the ms by 1000? How is overflow defined? It needs to strictly roll over, since that‘s what the timers expect.